vms/platformvm/txs/convert_subnet_tx_test.go

```diff
@@ -0,0 +1,635 @@
+	require.NoError(err)
+	require.Equal(`{
+	"networkID": 1,
+	"blockchainID": "11111111111111111111111111111111LpoYY",
+	"outputs": [
+		{
+			"assetID": "FvwEAhmxKfeiG8SnEvq42hc6whRyY3EFYAvebMqDNDGCgxN5Z",
+			"fxID": "spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ",
+			"output": {
+				"locktime": 87654321,
+				"output": {
+					"addresses": [],
+					"amount": 1,
+					"locktime": 12345678,
+					"threshold": 0
+				}
+			}
+		},
+		{
+			"assetID": "2Ab62uWwJw1T6VvmKD36ufsiuGZuX1pGykXAvPX1LtjTRHxwcc",
+			"fxID": "spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ",
+			"output": {
+				"locktime": 876543210,
+				"output": {
+					"addresses": [
+						"P-avax1g32kvaugnx4tk3z4vemc3xd2hdz92enh972wxr"
+					],
+					"amount": 18446744073709551615,
+					"locktime": 0,
+					"threshold": 1
+				}
+			}
+		}
+	],
+	"inputs": [
+		{
+			"txID": "2wiU5PnFTjTmoAXGZutHAsPF36qGGyLHYHj9G1Aucfmb3JFFGN",
+			"outputIndex": 1,
+			"assetID": "FvwEAhmxKfeiG8SnEvq42hc6whRyY3EFYAvebMqDNDGCgxN5Z",
+			"fxID": "spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ",
+			"input": {
+				"amount": 1000000000,
+				"signatureIndices": [
+					2,
+					5
+				]
+			}
+		},
+		{
+			"txID": "2wiU5PnFTjTmoAXGZutHAsPF36qGGyLHYHj9G1Aucfmb3JFFGN",
+			"outputIndex": 2,
+			"assetID": "2Ab62uWwJw1T6VvmKD36ufsiuGZuX1pGykXAvPX1LtjTRHxwcc",
+			"fxID": "spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ",
+			"input": {
+				"locktime": 876543210,
+				"input": {
+					"amount": 17293822569102704639,
+					"signatureIndices": [
+						0
+					]
+				}
+			}
+		},
+		{
+			"txID": "2wiU5PnFTjTmoAXGZutHAsPF36qGGyLHYHj9G1Aucfmb3JFFGN",
+			"outputIndex": 3,
+			"assetID": "2Ab62uWwJw1T6VvmKD36ufsiuGZuX1pGykXAvPX1LtjTRHxwcc",
+			"fxID": "spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ",
+			"input": {
+				"amount": 1152921504606846976,
+				"signatureIndices": []
+			}
+		}
+	],
+	"memo": "0xf09f98850a77656c6c2074686174277301234521",
+	"subnetID": "SkB92YpWm4UpburLz9tEKZw2i67H3FF6YkjaU4BkFUDTG9Xm",
+	"chainID": "NfebWJbJMmUpduqFCF8i1m5pstbVYLP1gGHbacrevXZMhpVMy",
+	"address": "0x000000000000000000000000000000000000dead",
+	"subnetAuthorization": {
+		"signatureIndices": []
+	}
+}`, string(unsignedComplexConvertSubnetTxJSONBytes))
+}
+
+func TestConvertSubnetTxSyntacticVerify(t *testing.T) {
+	var (
+		ctx         = snowtest.Context(t, ids.GenerateTestID())
+		validBaseTx = BaseTx{
+			BaseTx: avax.BaseTx{
+				NetworkID:    ctx.NetworkID,
+				BlockchainID: ctx.ChainID,
+			},
+		}
+		validSubnetID     = ids.GenerateTestID()
+		invalidAddress    = make(types.JSONByteSlice, MaxSubnetAddressLength+1)
+		validSubnetAuth   = &secp256k1fx.Input{}
+		invalidSubnetAuth = &secp256k1fx.Input{
+			SigIndices: []uint32{1, 0},
+		}
+	)
+
+	tests := []struct {
+		name        string
+		tx          *ConvertSubnetTx
+		expectedErr error
+	}{
+		{
+			name:        "nil tx",
+			tx:          nil,
+			expectedErr: ErrNilTx,
+		},
+		{
+			name: "already verified",
```

Comment:
If the tx were otherwise valid, we wouldn't be able to tell the difference between re-verifying it and caching that it was verified.

I'm open to suggestions here (ranging from alternative ways to verify this to just removing this check entirely)